va/va.go

```diff
@@ -464,32 +463,37 @@ func (va *ValidationAuthorityImpl) validateDNS01(ctx context.Context, identifier
 }
 
 func (va *ValidationAuthorityImpl) checkCAA(ctx context.Context, identifier core.AcmeIdentifier) *probs.ProblemDetails {
-	// Check CAA records for the requested identifier
-	present, valid, err := va.checkCAARecords(ctx, identifier)
-	if err != nil {
-		return bdns.ProblemDetailsFromDNSError(err)
+	var prob *probs.ProblemDetails
+	if va.caaClient != nil {
+		prob = va.checkCAAService(ctx, identifier)
+	} else {
+		present, valid, err := va.checkCAARecords(ctx, identifier)
+		if err != nil {
+			prob = bdns.ProblemDetailsFromDNSError(err)
+		} else {
+			// AUDIT[ Certificate Requests ] 11917fa4-10ef-4e0d-9105-bacbe7836a3c
+			va.log.AuditInfo(fmt.Sprintf(
+				"Checked CAA records for %s, [Present: %t, Valid for issuance: %t]",
+				identifier.Value,
+				present,
+				valid,
+			))
+			if !valid {
+				prob = probs.ConnectionFailure(fmt.Sprintf("CAA record for %s prevents issuance", identifier.Value))
```

Comment:
This seems like a place you would want to use your `CAAProblem` `ProblemType` by way of a `probs.CAAFailure` convenience method (see prev. comment).